apps/minifront/src/components/swap/swap-form/output/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Box } from '@penumbra-zone/ui/components/ui/box';
+import { AllSlices } from '../../../../state';
+import { useStoreShallow } from '../../../../utils/use-store-shallow';
+import { Input } from '@penumbra-zone/ui/components/ui/input';
+import { EstimateButton } from '../estimate-button';
+import { EstimatedOutputExplanation } from './estimated-output-explanation';
+import { motion } from 'framer-motion';
+
+const outputSelector = (state: AllSlices) => ({
+  assetOut: state.swap.assetOut,
+  minOutput: state.swap.dutchAuction.minOutput,
+  setMinOutput: state.swap.dutchAuction.setMinOutput,
+  maxOutput: state.swap.dutchAuction.maxOutput,
+  setMaxOutput: state.swap.dutchAuction.setMaxOutput,
+  estimateLoading: state.swap.dutchAuction.estimateLoading,
+  estimate: state.swap.dutchAuction.estimate,
+  estimateButtonDisabled:
+    state.swap.txInProgress || !state.swap.amount || state.swap.dutchAuction.estimateLoading,
+});
+
+export const Output = ({ layoutId }: { layoutId: string }) => {
+  const {
+    assetOut,
+    minOutput,
+    setMinOutput,
+    maxOutput,
+    setMaxOutput,
+    estimateLoading,
+    estimate,
+    estimateButtonDisabled,
+  } = useStoreShallow(outputSelector);
+
+  return (
+    <Box
+      layoutId={layoutId}
+      label='Output'
+      headerContent={
+        <EstimateButton
+          disabled={estimateButtonDisabled}
+          loading={estimateLoading}
+          onClick={() => void estimate()}
+        />
+      }
+    >
+      <motion.div layout className='flex flex-col gap-4'>
+        <motion.div layout className='flex max-w-[200px] grow flex-col gap-2'>
```

Comment:
Nested `<motion.div layout />`s allow the parent to animate its layout change without distorting the child.

Reply:
(Happy to explain in a demo on Thursday.)